.travis.yml

```diff
@@ -60,14 +12,90 @@ before_install:
   - chmod +x docker-compose
   - sudo mv docker-compose /usr/local/bin
 
-before_script:
-  - if [ "$CC" == "gcc" ]; then export CC=gcc-4.8; fi
-  - if [ "$CXX" == "g++" ]; then export CXX=g++-4.8; fi
-  - if [ "$CC" == "clang" ]; then export CC=clang-3.8; fi
-  - if [ "$CXX" == "clang++" ]; then export CXX=clang++-3.8; fi
-  - docker-compose build base
-  - if [ "$TARGET" == "android" ]; then docker-compose build java; fi
-  - docker-compose build ${TARGET}
-
-script:
-  - docker-compose run ${TARGET}
+jobs:
+  fast_finish: true
+  allow_failures:
+    - env: TARGET=sonar-scan
```

Comment:
Agree with you, actually Mraa sonar scans where failing because there where no "sonar.jsava.binearies" required property, the build was marked as success. What I see are this requirements:

1. Remove Allowed Failure
2. intel-iot-devkit/master branch must always run sonar scan, and fail the build is it not pass. If sonar found to many bugs, this will not fail the build. That failure will be reported in the sonar dashboard
3. "Internal" PRs must always run sonar scan, and fail the build is it not pass. If sonar found to many bugs, the PR will be marked as failure. 
4. For external PRs, The sonar-scan.sh already skip the scan. We cannot run sonar scan for this because the required env vars are not shared with external PRs. We could create a temporary internal PR to make sonar run the scan and get a preview of the results (this will be manual)
5. For forks, check if required variables are set, and skip the scan with a message explaining that if there do not exists. Otherwise, follow the same policy for intel-iot-devkit/master branch and internal PRs